### ISSUE TITLE
Backward Pass Panics `.with_padding_out(!= 0)`

### ISSUE THREAD

**Issue by TimerErTim** [tracel-ai/burn#4845]
**Describe the bug**
Activation function (sigmoid or tanh) into Conv1dTranspose with `padding_out != 0` causes the gradients to mismatch the output in the backward pass, leading to a panic during `loss.backward()`. This happens across all backends. I have not tested with other activation functions or constellations.

**To Reproduce**
```rust
#[derive(Module, Debug)]
pub struct DecoderModel<B: Backend> {
    pub linear_1: Linear<B>,
    pub linear_2: Linear<B>,
    pub activation: Sigmoid,
    pub unflatten_shape: [usize; 2], // [channels, length]
    pub convt1: ConvTranspose1d<B>,
    pub convt2: ConvTranspose1d<B>,
}

impl<B: Backend> DecoderModel<B> {
    /// latent: [batch, latent_space_features]
    /// returns [batch, 2, 50] (or same as input)
    pub fn forward(&self, input: Tensor<B, 2>) -> Tensor<B, 3> {
        let x = self.linear_1.forward(input);
        let x = self.activation.forward(x);
        let x = self.linear_2.forward(x);
        let bsz = x.shape()[0];
        let unflat = x.reshape([bsz, self.unflatten_shape[0], self.unflatten_shape[1]]);
        let x = self.convt1.forward(unflat);
        let x = self.activation.forward(x);
        let x = self.convt2.forward(x);
        Tensor::cat(vec![x.clone(), x.slice_dim(2, 1)], 2) // shape: [batch, 2, 50]
    }
}

#[derive(Config, Debug)]
pub struct AutoencoderModelConfig {
    pub book_depth: usize, // Should be 25 for this use case.
    pub latent_space_features: usize,
    pub encoder_channels_1: usize,
    pub encoder_channels_2: usize,
    pub encoder_kernel_size: usize,
    pub decoder_kernel_size: usize,
    pub pool_kernel: usize,
    pub dropout: f64,
}

impl AutoencoderModelConfig {
    pub fn init<B: Backend>(&self, device: &B::Device) -> OrderBookSeriesAutoencoderModel<B> {
        let flatten_dim = self.encoder_channels_2 * ((self.book_depth * 2 - (self.encoder_kernel_size - 1) * 2) / self.pool_kernel);
        OrderBookSeriesAutoencoderModel {
            encoder: EncoderModel {
                conv1: Conv1dConfig::new(2, self.encoder_channels_1, self.encoder_kernel_size)
                    .with_padding(burn::nn::PaddingConfig1d::Explicit(0))
                    .init(device),
                conv2: Conv1dConfig::new(self.encoder_channels_1, self.encoder_channels_2, self.encoder_kernel_size)
                    .with_padding(burn::nn::PaddingConfig1d::Explicit(0))
                    .init(device),
                activation: Sigmoid::new(),
                dropout: DropoutConfig::new(self.dropout).init(),
                pool: MaxPool1dConfig::new(self.pool_kernel).with_stride(self.pool_kernel).init(),
                flatten_dim,
                linear_1: LinearConfig::new(flatten_dim, flatten_dim / 2).init(device),
                linear_2: LinearConfig::new(flatten_dim / 2, self.latent_space_features).init(device),
            },
            decoder: DecoderModel {
                linear_1: LinearConfig::new(self.latent_space_features, flatten_dim / 2).init(device),
                linear_2: LinearConfig::new(flatten_dim / 2, flatten_dim).init(device),
                activation: Sigmoid::new(),
                unflatten_shape: [self.encoder_channels_2, (self.book_depth * 2 - (self.encoder_kernel_size - 1) * 2) / self.pool_kernel],
                convt1: ConvTranspose1dConfig::new([self.encoder_channels_2, self.encoder_channels_1], self.decoder_kernel_size)
                    .with_padding(0)
                    .with_stride(self.pool_kernel)
                    .init(device),
                convt2: ConvTranspose1dConfig::new([self.encoder_channels_1, 2], self.decoder_kernel_size)
                    .with_padding(0)
                    //.with_stride(2)
                    .with_padding_out(0)
                    .init(device),
            },
        }
    }
}
```

**Expected behavior**
The backward pass should work properly and reliably, as it should granted the forward pass works.

**Desktop (please complete the following information):**
 - OS: Linux, openSUSE, Kernel 6.19.12-1-default
 - Browser: Headless Server
 - Version: 0.20.1

**Additional context**
Confirmed by having no problem (apart of curse of dimensionality) when using `.with_padding_out(0)`.
